src/Serializer/AbstractConstraintViolationListNormalizer.php

```diff
@@ -45,6 +45,13 @@ public function supportsNormalization(mixed $data, string $format = null, array
     {
         return static::FORMAT === $format && $data instanceof ConstraintViolationListInterface;
     }
+    
+    public function getSupportedTypes(?string $format)
+    {
+        return [
+            ConstraintViolationListInterface::class => $this->hasCacheableSupportsMethod(),
+        ];
+    }
```

Comment:
I would recommend taking the format into account to avoid unwanted side effects and improve the performance event further, how about this one:

```suggestion
    /**
     * @return array<class-string<ConstraintViolationListInterface::class>, bool>|list<mixed>
     */
    public function getSupportedTypes(?string $format)
    {
        return match($format) {
            static::FORMAT => [
                ConstraintViolationListInterface::class => $this->hasCacheableSupportsMethod(),
            ],
            default => [];
        }
    }
```

Also since the `CacheableSupportsMethodInterface` is also deprecated it might be better to avoid right out of the gate and directly change it to:

```suggestion
    /**
     * @return array<class-string<ConstraintViolationListInterface::class>, true>|list<mixed>
     */
    public function getSupportedTypes(?string $format) {
        return match($format) {
            static::FORMAT => [
                ConstraintViolationListInterface::class => true,
            ],
            default => [];
        }
    }
```

Reply:
For a moment, I thought of deleting the "hasCacheableSupportsMethod", but I was afraid it would cause backward compatibility problems.

Yes, since the "CacheableSupportsMethodInterface" interface has become deprecated, I suppose that's the wisest thing to do.

Reply:
This is not exactly what I meant, in the first iteration we should add the getSupportedTypes method to handle the deprecation from Symfony 6.3.

We cannot directly drop this method and interface as this would break with < Symfony 6.3.

I'am not entirely sure how this deprecation has to be handled inside API Platform to keep the backward compatibility, maybe a new minor version is required which indicates a conflict with Symfony <6.3 to be able to drop the usage of the "CacheableSupportsMethodInterface".

I'am happy to here a suggestion from the maintainers.